core/src/services/webhdfs/backend.rs

```diff
@@ -195,47 +195,72 @@ pub struct WebhdfsBackend {
 }
 
 impl WebhdfsBackend {
-    /// create object or make a directory
-    ///
-    /// TODO: we should split it into mkdir and create
-    pub fn webhdfs_create_object_request(
+    pub fn webhdfs_create_dir_request(&self, path: &str) -> Result<Request<AsyncBody>> {
+        let p = build_abs_path(&self.root, path);
+
+        let mut url = format!(
+            "{}/webhdfs/v1/{}?op=MKDIRS&overwrite=true&noredirect=true",
+            self.endpoint,
+            percent_encode_path(&p),
+        );
+        if let Some(auth) = &self.auth {
+            url += format!("&{auth}").as_str();
+        }
+
+        let req = Request::put(&url);
+
+        req.body(AsyncBody::Empty).map_err(new_request_build_error)
+    }
+    /// create object
+    pub async fn webhdfs_create_object_request(
         &self,
         path: &str,
-        size: Option<usize>,
+        size: Option<u64>,
         args: &OpWrite,
         body: AsyncBody,
     ) -> Result<Request<AsyncBody>> {
         let p = build_abs_path(&self.root, path);
-        let op = if path.ends_with('/') {
-            "MKDIRS"
-        } else {
-            "CREATE"
-        };
+
         let mut url = format!(
-            "{}/webhdfs/v1/{}?op={}&overwrite=true",
+            "{}/webhdfs/v1/{}?op=CREATE&overwrite=true&noredirect=true",
             self.endpoint,
             percent_encode_path(&p),
-            op,
         );
         if let Some(auth) = &self.auth {
             url += format!("&{auth}").as_str();
         }
 
-        let mut req = Request::put(&url);
+        let req = Request::put(&url);
 
-        // mkdir does not redirect
-        if path.ends_with('/') {
-            return req.body(AsyncBody::Empty).map_err(new_request_build_error);
-        }
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
 
-        if let Some(size) = size {
-            req = req.header(CONTENT_LENGTH, size.to_string());
-        }
-        if let Some(content_type) = args.content_type() {
-            req = req.header(CONTENT_TYPE, content_type);
-        }
+        let resp = self.client.send(req).await?;
 
-        req.body(body).map_err(new_request_build_error)
+        let status = resp.status();
+        match status {
+            StatusCode::CREATED | StatusCode::OK => {
```

Comment:
Putting complex logic in a match clause hinders readability.

How about write like this:

```rust
if status != StatusCode::CREATED && status != StatusCode::OK {
    return Err(parse_error(resp).await?);
}

// other logic
```